compiler/src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -454,6 +454,8 @@ object Erasure extends TypeTestsCasts{
       val Apply(fun, args) = tree
       if (fun.symbol == defn.cbnArg)
         typedUnadapted(args.head, pt)
+      else if (fun.symbol eq defn.PhantomAssume)
+        ref(defn.BoxedUnit_UNIT)
```

Comment:
Same issue: need to document why we need the erasure of this. Is it just to get signatures or is it something else?